stacker/actions/diff.py

```diff
@@ -215,11 +215,11 @@ def _diff_stack(self, stack, **kwargs):
 
         stack.resolve(self.context, self.provider)
         # generate our own template & params
-        new_template = stack.blueprint.to_json()
         parameters = self.build_parameters(stack)
         new_params = dict()
         for p in parameters:
             new_params[p['ParameterKey']] = p['ParameterValue']
+        new_template = stack.blueprint.to_json(new_params)
```

Comment:
I'm not sure this is the correct behavior. `to_json` takes a dict of variables, not necessarily parameters. This change would make the behavior in `_diff_stack` significantly different from [_launch_stack](https://github.com/remind101/stacker/blob/master/stacker/actions/build.py#L270-L276). I think it's important that these use similar code paths or we're gonna run into these types of problems.

I think the root of the problem is the change from `blueprint.rendered` to `blueprint.to_json` in https://github.com/remind101/stacker/commit/4d0fad8665485199163abccc22919f9ef2b65343#diff-1a5240c955da6d0e576a399c23c44953L215. `to_json` apparently has too many side effects to really rely on it here.

I think this should probably just be switched back to using `rendered`, which is what we use in `_launch_stack`. It would mean diffing a JSON template against a YAML template will be garbage, but I think that's fine. Thoughts @troyready?

Reply:
I think calling `.rendered` sounds like right thing to do. (bfe0343)

RE: diffing YAML against JSON. On a related note, I was just talking to a coworker yesterday about adding support to stacker to use YAML as the final representation instead of JSON. IMO, diffs of yaml are probably a bit cleaner, anyway. I'd be happy to work on adding that as a config option, if others thought it would be useful.